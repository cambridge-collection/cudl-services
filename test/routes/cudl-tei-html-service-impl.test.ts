import {
  TeiHtmlServiceContent,
  teiHtmlServicePathGenerator,
} from '../../src/routes/cudl-tei-html-service-impl';
import {product} from '../utils';

describe('teiTranscriptionPathGenerator', () => {
  test.each([
    [
      TeiHtmlServiceContent.TRANSCRIPTION,
      {id: 'MS-ADD-10067', start: 'i11'},
      'html/data/tei/MS-ADD-10067/MS-ADD-10067-i11.html',
    ],
    [
      TeiHtmlServiceContent.TRANSCRIPTION,
      {id: 'MS-ADD-10067', start: 'i11', end: 'i11'},
      'html/data/tei/MS-ADD-10067/MS-ADD-10067-i11.html',
    ],
    [
      TeiHtmlServiceContent.TRANSCRIPTION,
      {id: 'MS-DAR-00115-00078-A', start: 'i1', end: 'i8'},
      'html/data/tei/MS-DAR-00115-00078-A/MS-DAR-00115-00078-A-i1-i8.html',
    ],
    [
      TeiHtmlServiceContent.TRANSLATION,
      {id: 'MS-ADD-10067', start: 'i11'},
      'html/data/tei/MS-ADD-10067/MS-ADD-10067-i11-translation.html',
    ],
    [
      TeiHtmlServiceContent.TRANSLATION,
      {id: 'MS-ADD-10067', start: 'i11', end: 'i11'},
      'html/data/tei/MS-ADD-10067/MS-ADD-10067-i11-translation.html',
    ],
    [
      TeiHtmlServiceContent.TRANSLATION,
      {id: 'MS-DAR-00115-00078-A', start: 'i1', end: 'i8'},
      'html/data/tei/MS-DAR-00115-00078-A/MS-DAR-00115-00078-A-i1-i8-translation.html',
    ],
  ])(
    'for type %j and params %j generates path %j',
    (type, params, expected) => {
      expect(teiHtmlServicePathGenerator(type)({params})).toEqual(expected);
    }
  );

  test.each(
    Array.from(
      product<TeiHtmlServiceContent, Record<string, string>>(
        [
          TeiHtmlServiceContent.TRANSCRIPTION,
          TeiHtmlServiceContent.TRANSLATION,
        ],
        [{id: 'i', end: 'e'}, {id: 'i'}, {start: 's', end: 'e'}, {start: 's'}]
      )
    )
  )(
    'for type %j throws with missing param error when given params: %j',
    (type, params) => {
      expect(() =>
        teiHtmlServicePathGenerator(type)({params})
      ).toThrowErrorMatchingSnapshot();
    }
  );
});
