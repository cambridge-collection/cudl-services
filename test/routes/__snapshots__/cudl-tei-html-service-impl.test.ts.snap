// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teiTranscriptionPathGenerator for type "transcription" throws with missing param error when given params: {"id":"i","end":"e"} 1`] = `"failed to generate TEI HTML service transcription path: id and start params must be present"`;

exports[`teiTranscriptionPathGenerator for type "transcription" throws with missing param error when given params: {"id":"i"} 1`] = `"failed to generate TEI HTML service transcription path: id and start params must be present"`;

exports[`teiTranscriptionPathGenerator for type "transcription" throws with missing param error when given params: {"start":"s","end":"e"} 1`] = `"failed to generate TEI HTML service transcription path: id and start params must be present"`;

exports[`teiTranscriptionPathGenerator for type "transcription" throws with missing param error when given params: {"start":"s"} 1`] = `"failed to generate TEI HTML service transcription path: id and start params must be present"`;

exports[`teiTranscriptionPathGenerator for type "translation" throws with missing param error when given params: {"id":"i","end":"e"} 1`] = `"failed to generate TEI HTML service translation path: id and start params must be present"`;

exports[`teiTranscriptionPathGenerator for type "translation" throws with missing param error when given params: {"id":"i"} 1`] = `"failed to generate TEI HTML service translation path: id and start params must be present"`;

exports[`teiTranscriptionPathGenerator for type "translation" throws with missing param error when given params: {"start":"s","end":"e"} 1`] = `"failed to generate TEI HTML service translation path: id and start params must be present"`;

exports[`teiTranscriptionPathGenerator for type "translation" throws with missing param error when given params: {"start":"s"} 1`] = `"failed to generate TEI HTML service translation path: id and start params must be present"`;
